File: internal/bindings/connection.go
package bindings

import (
	"gomyadm/internal/models"
	"gomyadm/internal/services"
)

type ConnectionBinding struct {
	service *services.ConnectionsStore
}

func NewConnection(s *services.ConnectionsStore) *ConnectionBinding {
	return &ConnectionBinding{service: s}
}

func (a *ConnectionBinding) GetAll() []models.ConnectionConfig {
	return a.service.GetAll()
}

func (a *ConnectionBinding) Create(con models.ConnectionConfig) error {
	return a.service.Create(con)
}

func (a *ConnectionBinding) Update(id string, con models.ConnectionConfig) error {
	return a.service.Update(id, con)
}

func (a *ConnectionBinding) Delete(id string) error {
	return a.service.Delete(id)
}

File: frontend/src/repositories/wails/connection.ts
import type { Connection } from '@/models';
import { GetAll, Create, Update, Delete } from '../../../wailsjs/go/bindings/ConnectionBinding'
import type { ConnectionRepository } from "../connection";
import type { Result } from '..';

export class WailsConnectionRepository implements ConnectionRepository {
  async getAll(): Promise<Result<Connection[]>> {
    try {
      return {
        ok: true, data: await GetAll()
      }
    } catch (error) {
      return {
        ok: false, error: error instanceof Error ? error.message : "Erro ao conectar"
      }
    }
  }

  async create(con: any): Promise<Result<void>> {
    try {
      await Create(con)
      return {
        ok: true, data: undefined
      }
    } catch (error) {
      return {
        ok: false, error: error instanceof Error ? error.message : "Erro ao conectar"
      }
    }
  }

  async update(id: string, con: any): Promise<Result<void>> {
    try {
      await Update(id, con)
      return {
        ok: true, data: undefined
      }
    } catch (error) {
      return {
        ok: false, error: error instanceof Error ? error.message : "Erro ao conectar"
      }
    }
  }

  async delete(id: string): Promise<Result<void>> {
    try {
      await Delete(id)
      return {
        ok: true, data: undefined
      }
    } catch (error) {
      return {
        ok: false, error: error instanceof Error ? error.message : "Erro ao conectar"
      }
    }
  }
}